### DISCUSSION TITLE
Dispatching event from mount function aren't working

### DISCUSSION THREAD

**Discussion by sakjain02** [livewire/livewire#5617] [1. Help] [answered]
public function mount()
    {
        $this->dispatchBrowserEvent('show-modal');
    }
    
    Above event doesn't work
    window.addEventListener('show-modal', event => {
        $('#login-signup').modal('show');
    });
    
Same code works on click or change event but not working in mount function.

Livewire version: ^2.3
Laravel version: ^9.0

Tried every pinch of solution, nothing seems to be working. It might be a bug as it doesn't provide any error message or log as well.

**Comment by joshhanley** [comment, marked as answer]
@sakjain02 Livewire doesn't support emitting events in the mount method as Livewire's javascript isn't running yet. You could try doing it in a `wire:init` call. https://laravel-livewire.com/docs/2.x/defer-loading

Hope this helps!

**Reply by sakjain02** [reply]
Yeah it works. Thanks for the help 👍
